### PR TITLE
Add Update function for Employees

### LINKED ISSUE #20: Edit employee data
# REQUIREMENTS
Create a function that searches for an employee in the database by means of his ID and displays the data of the employee found with the option to edit it.

# Acceptance Criteria
- [x] Employee Finder
- [x] Function Edit Employee

### PR DESCRIPTION
**Pull request by SampabloC**
- Add update function on employees controller
 - Modify _list view to add buttons show, update and delete on every employee.
 - Remove employees from root directory.
 - Add _form to put in the formulary to use in new and edit view.

Co-authored-by: RobertCarMed <betocarrasco3652@gmail.com>
Co-authored-by: mrivh <m.alicia.riv@gmail.com>

Resolve #20

### CODE REVIEW

File: app/controllers/employees_controller.rb [hunk 1]
```diff
@@ -11,27 +11,33 @@ def new
   end
 
   def show
-
+    @employees = Employee.find(params[:id])
```
Comment (juliocabrera820) [inline app/controllers/employees_controller.rb:14]: you can use a hook, in this case before_action, you could create a private method called set_employee like this:
```ruby
def set_employee
  @employee = Employee.find(params[:id])
end
```

File: app/controllers/employees_controller.rb [hunk 2]
```diff
@@ -11,27 +11,33 @@ def new
...
+    employee = Employee.new(employee_params)
+    employee.administrator_id = "1"
     employee.save
-
-    redirect_to root_path
+    redirect_to employee_path
   end
 
   def edit
+    @employees = Employee.find(params[:id])
```
Comment (juliocabrera820) [inline app/controllers/employees_controller.rb:25]: you can use a hook, in this case before_action, you could create a private method called set_employee like this:
```ruby
def set_employee
  @employee = Employee.find(params[:id])
end
```

File: app/controllers/employees_controller.rb [hunk 3]
```diff
@@ -11,27 +11,33 @@ def new
...
+  end
+
+  def update
+    @employees = Employee.find(params[:id])
```
Comment (juliocabrera820) [inline app/controllers/employees_controller.rb:29]: you can use a hook, in this case before_action, you could create a private method called set_employee like this:
```ruby
def set_employee
  @employee = Employee.find(params[:id])
end
```

File: app/controllers/employees_controller.rb [hunk 4]
```diff
@@ -11,27 +11,33 @@ def new
...
+
+  def destroy; end
+  
+  private
 
+  def employee_params
+    params.require(:employee).permit(:employeeNumber, :nameEmployee, :emailEmployee, :positionEmployee, :attendanceDate)#, :administrator_id)
   end
+
+
```
Comment (juliocabrera820) [inline app/controllers/employees_controller.rb:42]:
```ruby
def set_employee
  @employee = Employee.find(params[:id])
end
```

Comment (juliocabrera820) [inline app/controllers/employees_controller.rb:42]:
```ruby
def set_employee
  @employee = Employee.find(params[:id])
end
```

**Review by SEVERO96 — APPROVED** [review]

File: app/views/employees/_list.html.erb [hunk 5]
```diff
@@ -35,6 +38,17 @@
...
           <td>
-            <% end %>
+            <%= link_to "Show", employee_path(employee), class: "btn btn-info" %>
+          </td>
+          <td>
+            <%= link_to "Edit", edit_employee_path(employee), class: "btn btn-warning" %>
+          </td>
+          
+          <td>
+            <%= link_to "Delete", employee_path(employee) ,class: "btn btn-danger" %>
```
Comment (xFerchoVJ) [inline app/views/employees/_list.html.erb:48]: Missing space after comma
but everyhting looks good to me!
`<%= link_to "Delete", employee_path(employee), class: "btn btn-danger" %>`

File: config/routes.rb [hunk 6]
```diff
@@ -5,5 +5,5 @@
   get '/administrators/manageCompany', to: 'administrators#manageCompany'
   get '/administrators/reports', to: 'administrators#reports'
   resources :employees, :administrators
-  root 'employees#index'
+  #root 'employees#index'
```
Comment (juliocabrera820) [inline config/routes.rb:8]: commented-out code must be removed

**Review by juliocabrera820 — APPROVED** [review]
LGTM 👍🏼

File: app/controllers/employees_controller.rb [hunk 7]
```diff
@@ -11,27 +11,33 @@ def new
...
+    @employees = Employee.find(params[:id])
   end
 
   def create
-    data = params.require(:employee)
-    employee = Employee.new
-    employee.nameEmployee = data["nameEmployee"]
-    employee.emailEmployee = data["emailEmployee"]
-    employee.positionEmployee = data["positionEmployee"]
-    employee.attendanceDate = data["attendanceDate"]
```
Comment (xFerchoVJ) [inline app/controllers/employees_controller.rb:23]: You should use snake case with your employee attributes
Example:
name_employee

**Review by xFerchoVJ — CHANGES_REQUESTED** [review]

File: app/views/employees/_list.html.erb (hunk 5 above)
Comment (SampabloC) [reply app/views/employees/_list.html.erb:48]: ok, i will correct that little mistake

File: app/controllers/employees_controller.rb (hunk 7 above)
Comment (SampabloC) [reply app/controllers/employees_controller.rb:23]: Sure, but i need to modify some files including my tables, we will planned to do that on a new issue

**Review by SaraNavarro — APPROVED** [review]

**Review by xFerchoVJ — APPROVED** [review]

**Comment by xFerchoVJ** [comment]
LGTM!!!

**Pull request merged**